federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -747,24 +736,9 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 			return
 		}
 		desiredIngress := &extensionsv1beta1.Ingress{}
-		objMeta, err := api.Scheme.DeepCopy(&baseIngress.ObjectMeta)
-		if err != nil {
-			glog.Errorf("Error deep copying ObjectMeta: %v", err)
-		}
-		objSpec, err := api.Scheme.DeepCopy(&baseIngress.Spec)
-		if err != nil {
-			glog.Errorf("Error deep copying Spec: %v", err)
-		}
-		objMetaCopy, ok := objMeta.(*metav1.ObjectMeta)
-		if !ok {
-			glog.Errorf("Internal error: Failed to cast to *metav1.ObjectMeta: %v", objMeta)
-		}
-		desiredIngress.ObjectMeta = *objMetaCopy
-		objSpecCopy, ok := objSpec.(*extensionsv1beta1.IngressSpec)
-		if !ok {
-			glog.Errorf("Internal error: Failed to cast to extensionsv1beta1.Ingressespec: %v", objSpec)
-		}
-		desiredIngress.Spec = *objSpecCopy
+		baseIngress.ObjectMeta.DeepCopyInto(&desiredIngress.ObjectMeta)
```

Comment:
Based on the old code, we don't.  I think the assignment is more clear to readers.  I'd pay for the allocation unless we find a problem.